packages/next/src/build/webpack/plugins/parallel-acorn/acorn-parser-worker.ts

```diff
@@ -0,0 +1,29 @@
+import { Parser as AcornParser } from 'next/dist/compiled/acorn'
+import { importAssertions } from 'acorn-import-assertions'
+const parser = AcornParser.extend(importAssertions)
+
+const defaultParserOptions = {
+  ranges: true,
+  locations: true,
+  ecmaVersion: 'latest',
+  sourceType: 'module',
+  // https://github.com/tc39/proposal-hashbang
+  allowHashBang: true,
+  onComment: null,
+}
+
+export function parseModuleWebpackAST(
+  sourceType: string | undefined,
+  source: any
+) {
+  const type = sourceType ? sourceType : 'module'
+
+  const parserOptions = {
+    ...defaultParserOptions,
+    allowReturnOutsideFunction: type === 'script',
+    // ...options,
+    sourceType: type === 'auto' ? 'module' : type,
+  }
+
+  return parser.parse(source, parserOptions)
```

Comment:
Not for this PR but random idea:

swc internally have `compat` mode that generates estree/acron compatible ast from swc's ast, it'd be interesting if that could bring some perf improvement. (of course if it's fully compatible)

Reply:
@kwonoj that's something @kdy1 worked on but we couldn't use it because of compat issues and deprioritized the work to focus on minification / turbopack instead 👍

Reply:
Oh I see, thanks for the context 🙇